### ISSUE TITLE
Planned Analysis: Co-Occurence / Mutual Exclusivity 

### ISSUE THREAD

**Issue by PichaiRaman** [AlexsLemonade/OpenPBTA-analysis#13]
Determining genetic lesions (Mutation, CNV, Fusion) and/or pathways which co-occur or are mutually exclusive across the PBTA. This could help associate lesions with pathways or define potential synthetic lethality relationship.

**Comment by cansav09** [comment]
Would we be looking as an end goal to make some figure like this where we would likely have disease type on the bottom? ![10 1038_ng1975-f2_large_2](https://user-images.githubusercontent.com/23458084/61900827-67e3ad80-aeec-11e9-82ea-a959d85b4761.jpg) (from Rotika et al, 2019).

**Comment by cansav09** [comment]
Would we be looking to have something like this figure for this analysis, but with the disease type along the bottom? ![Screen Shot 2019-07-25 at 3 01 05 PM](https://user-images.githubusercontent.com/23458084/61901156-23a4dd00-aeed-11e9-9d69-d17c405bf5fb.png) This is from Rotika et al, 2019 doi: http://dx.doi.org/10.1101/566455.

**Comment by cansav09** [comment]
And after #19 is done, the molecular subtypes could also be added.

**Comment by cansav09** [comment]
One of my initial questions is how we'd like to combine results that measure similar things. 

For example, we have two fusions results. Do we want only take the common fusions from both files?  
```
arriba.fusions.tsv
star-fusion.fusions.tsv
```
And if I understand correctly, these all have results that overlap: 
```
strelka2.maf
manta-sv.maf
mutect2.maf
```
Note that I'm still digging into all these files and determining what's here, so after I do some initial analyses I may come back with a suggestion on how to handle this, but if someone knows this data better and has an idea of what we want to do first, then I'm of course open to those suggestions.

**Comment by jharenza** [comment]
> One of my initial questions is how we'd like to combine results that measure similar things.
> 
> For example, we have two fusions results. Do we want only take the common fusions from both files?
> 
> ```
> arriba.fusions.tsv
> star-fusion.fusions.tsv
> ```
> 
> And if I understand correctly, these all have results that overlap:
> 
> ```
> strelka2.maf
> manta-sv.maf
> mutect2.maf
> ```
> 
> Note that I'm still digging into all these files and determining what's here, so after I do some initial analyses I may come back with a suggestion on how to handle this, but if someone knows this data better and has an idea of what we want to do first, then I'm of course open to those suggestions.

We actually have created a high-confidence set of calls by integrating the two fusion callers - will add that progress to #10 soon! The same should be done with the SNVs and SVs (Lumpy SV data yet to come).

**Comment by cansav09** [comment]
Okay. So I should hold off on doing this until those integrated results come back?

**Comment by jharenza** [comment]
> Okay. So I should hold off on doing this until those integrated results come back?

I think you can integrate the SNV results and do the mutual exclusivity/co-occurrance analysis on SNVs more immediately!

**Comment by cansav09** [comment]
Okay. How should I integrate those? Sounds like you guys have already done some work on this?

**Comment by jharenza** [comment]
> Would we be looking to have something like this figure for this analysis, but with the disease type labels? ![Screen Shot 2019-07-25 at 3 01 05 PM](https://user-images.githubusercontent.com/23458084/61901156-23a4dd00-aeed-11e9-9d69-d17c405bf5fb.png) This is from Rotika et al, 2019 doi: http://dx.doi.org/10.1101/566455.

This figure is related to #6. For this issue, we would look for something similar to
[example-plot.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/3433544/example-plot.pdf) (created with maftools for another solid tumor dataset I had) for specific histologies or another mode of visualizing/tabling statistical testing for these mutation relationships. First, discover those relationships, then summarize. May depend on #19, but can give a go with broader histologies or start with one, for example, Medulloblastoma, High-grade glioma, low-grade glioma.

**Comment by jharenza** [comment]
> Okay. How should I integrate those? Sounds like you guys have already done some work on this?

We do not yet have this automated, so you can come up with the total merge of mutations per sample based on the two mutation algorithms, Mutect2 and Strelka2 and then start with those. It may also be a good idea to investigate mutations present in only one algorithm for potential artifacts (some will be real), but this may be constitute another issue and may cause these issues dependency on that - thoughts, @cgreene ?

**Comment by cansav09** [comment]
Sounds like I should do some initial analyses to see how much overlap there is between Mutext2 and Strelka2 and then I'll report back and we can try to make some further decisions. 
I will hold off for now on combining the fusions results since it sounds like [you are working on this](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/10#issuecomment-515178814.

**Comment by jharenza** [comment]
> Sounds like I should do some initial analyses to see how much overlap there is between Mutext2 and Strelka2 and then I'll report back and we can try to make some further decisions.
> I will hold off for now on combining the fusions results since it sounds like [you are working on this](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/10#issuecomment-515178814).

**Comment by cgreene** [comment]
Maybe it'd be good to add a new issue for:
> Evaluate concordance between Mutext2 and Strelka2 and decide on next steps.

This may be helpful to keep discussion within the issue focused on a single topic.

**Comment by jharenza** [comment]
created #30

**Comment by jashapiro** [comment]
I'm going to start working on this, aiming to build a figure similar to the one @jharenza presented: 
![example-plot](https://user-images.githubusercontent.com/20483/66420235-875e8280-e9d3-11e9-8f77-7b8922eece75.png)

For now I will used the strelka2 data, but it should be flexible enough to feed in whatever final set of data we would be interested in. 

I was planning to allow production alternative plots for different VAF cutoffs, as well as grouping by gene, filtering by effect, mutation type, etc.

**Comment by jashapiro** [comment]
I'm currently working on adding processing of CNV outputs for these analyses. The current plan is to make plots with SNV only, CNV only, and SNV + CNV. For initial analysis, I am taking each gene + loss or gain status (broadly interpreted) as the mutation unit for each sample. However, there are a few issues worth discussing before finalizing the analysis.

* In the current analysis (CNVkit), almost every gene has gain or loss in at least one sample. I expect this to be improved by #128, but on other part of the issue may be the use of broadly defined gene locations in #186/#195. Restricting those calls to exons may also improve things.

* Large losses or gains will necessarily result in many highly correlated CNV calls. How to display these remains an open question, and I would welcome any thoughts or suggestions. This will likely wait on conclusion of #186.

**Comment by jaclyn-taroni** [comment]
Closing all planned analysis tickets in favor of opening new proposed analysis/updated analysis tickets as needed.
